content-src/components/Header/Header.js

```diff
@@ -33,6 +36,13 @@ const Header = React.createClass({
         </ul>
       </section>
       <section className="spacer" />
+      <section className="filter">
+        <input
+          onChange={e => props.dispatch(actions.NotifyFilterQuery(e.target.value))}
```

Comment:
@piatra what do you mean by debounce?
and also nah i don't think you need to declare dispatch as a required prop